### PR TITLE
feat(FEC-12450): build image uri based on redirected uri

### PR DESCRIPTION
**Pull request by giladna**

### CODE REVIEW

File: src/parser/dash-manifest-parser.js [hunk 1]
```diff
@@ -50,6 +56,13 @@ class DashManifestParser {
   }
 
   _parseAdaptionSets = () => {
+    const baseURL = XmlUtils.findElements(this._xmlDoc, MpdUtils.TagTypes.BASE_URL);
+    if (baseURL && baseURL.length > 0) {
```
Comment (SivanA-Kaltura) [inline src/parser/dash-manifest-parser.js:60]: looks like it always returns an array
https://developer.mozilla.org/en-US/docs/Web/API/Element/getElementsByTagName

File: src/parser/dash-manifest-parser.js [hunk 2]
```diff
@@ -50,6 +56,13 @@ class DashManifestParser {
...
+      if (baseURL[0].innerHTML) {
+        this._baseURL = baseURL[0].textContent;
+        console.log(baseURL[0].textContent);
```
Comment (SivanA-Kaltura) [inline src/parser/dash-manifest-parser.js:63]: remove and/or replace with an info log

File: src/parser/parser-utils.js [hunk 3]
```diff
@@ -48,12 +48,14 @@ const ParserUtils = {
...
 const MpdUtils = {
   TagTypes: {
+    BASE_URL: 'BaseURL',
     SEGMENT_TEMPLATE: 'SegmentTemplate',
     REPRESENTATION: 'Representation',
     ESSENTIAL_PROPERTY: 'EssentialProperty',
     ADAPTATION_SET: 'AdaptationSet'
   },
   AttributeTypes: {
+    BASE_URL: 'BaseURL',
```
Comment (SivanA-Kaltura) [inline src/parser/parser-utils.js:58]: is this used ?

File: src/dash-adapter.js [hunk 4]
```diff
@@ -753,6 +753,7 @@ export default class DashAdapter extends BaseMediaSourceAdapter {
           break;
         case shaka.net.NetworkingEngine.RequestType.MANIFEST:
           this._parseManifest(response.data);
+          this._finalUri = response.uri;
```
Comment (SivanA-Kaltura) [inline src/dash-adapter.js:756]: why final ?

File: src/dash-adapter.js [hunk 5]
```diff
@@ -1057,8 +1058,9 @@ export default class DashAdapter extends BaseMediaSourceAdapter {
   _getParsedImageTracks(): Array<ImageTrack> {
     if (this._manifestParser) {
       const imageSet = this._manifestParser.getImageSet();
+      const middleBaseURL = this._manifestParser.getBaseUrl();
```
Comment (SivanA-Kaltura) [inline src/dash-adapter.js:1061]: middleBaseUrl is a string that's being inserted into the middle of a url ? what does it represent ? would it usually be empty when we don't use redirect ?

**Review by SivanA-Kaltura — CHANGES_REQUESTED** [review]

File: src/parser/dash-manifest-parser.js (hunk 1 above)
Comment (giladna) [reply src/parser/dash-manifest-parser.js:60]: @SivanA-Kaltura  what is required here?

**Review by SivanA-Kaltura — APPROVED** [review]

**Pull request merged**